extension/src/popup/ducks/transactionSubmission.ts

```diff
@@ -260,6 +276,27 @@ const transactionSubmissionSlice = createSlice({
         assetIcons,
       };
     });
+    builder.addCase(getBestPath.fulfilled, (state, action) => {
+      if (!action.payload) {
+        state.transactionData.path = [];
+        state.transactionData.destinationAmount = "";
+        return;
+      }
+
+      // store in canonical form for easier use
+      const path = [];
+      for (let i = 0; i < action.payload.path.length; i += 1) {
+        const p = action.payload.path[i];
+        path.push(
+          p.asset_code === "native"
+            ? "native"
+            : `${p.asset_code}:${p.asset_issuer}`,
+        );
```

Comment:
you can use the `getCanonicalFromAsset` helper here :)
extension/src/helpers/stellar.ts

Reply:
https://github.com/stellar/freighter/pull/413/commits/832e95fcf0a3ed782b57b23d0d6078e11d07006c